lib/helpers.py

```diff
@@ -1,323 +1,323 @@
-                try:
-                    nonascii = string.encode('raw_unicode_escape')
-                except Exception, e:
-                    nonascii = str("%s" % string)
-
-        except Exception as e:
-            # traceback.print_exc()
-            # print "All methods failed - removing characters"
-            # Generate a new string without disturbing characters
-            nonascii = "".join(i for i in string if ord(i)<127 and ord(i)>31)
-
-    except Exception as e:
-        traceback.print_exc()
-        pass
-
-    return nonascii
-
-def getAge(filePath):
-    try:
-        stats=os.stat(filePath)
-
-        # Created
-        ctime=stats.st_ctime
-        # Modified
-        mtime=stats.st_mtime
-        # Accessed
-        atime=stats.st_atime
-
-    except Exception as e:
-        # traceback.print_exc()
-        return (0, 0, 0)
-
-    # print "%s %s %s" % ( ctime, mtime, atime )
-    return (ctime, mtime, atime)
-
-def getAgeString(filePath):
-    ( ctime, mtime, atime ) = getAge(filePath)
-    timestring = ""
-    try:
-        timestring = "CREATED: %s MODIFIED: %s ACCESSED: %s" % ( time.ctime(ctime), time.ctime(mtime), time.ctime(atime) )
-    except Exception as e:
-        timestring = "CREATED: not_available MODIFIED: not_available ACCESSED: not_available"
-    return timestring
-
-
-def runProcess(command, timeout=10):
-    """
-    Run a process and check it's output
-    :param command:
-    :return output:
-    """
-    output = ""
-    returnCode = 0
-
-    # Kill check
-    kill_check = threading.Event()
-    def _kill_process_after_a_timeout(pid):
-        os.kill(pid, signal.SIGTERM)
-        kill_check.set() # tell the main routine that we had to kill
-        print("timeout hit - killing pid {0}".format(pid))
-        # use SIGKILL if hard to kill...
-        return "", 1
-    try:
-        p = subprocess.Popen(command, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
-    except subprocess.CalledProcessError as e:
-        returnCode = e.returncode
-        traceback.print_exc()
-    #print p.communicate()[0]
-    pid = p.pid
-    watchdog = threading.Timer(timeout, _kill_process_after_a_timeout, args=(pid, ))
-    watchdog.start()
-    (stdout, stderr) = p.communicate()
-    output = "{0} {1}".format(stdout, stderr)
-    watchdog.cancel() # if it's still waiting to run
-    success = not kill_check.isSet()
-    kill_check.clear()
-
-    return output, returnCode
-
-def getHostname(os_platform):
-    """
-    Generate and return a hostname
-    :return:
-    """
-    # Computername
-    if os_platform == "linux" or os_platform == "osx":
-        return os.uname()[1]
-    else:
+#!/usr/bin/env python
+# -*- coding: iso-8859-1 -*-
+# -*- coding: utf-8 -*-
+#  Python3 Support
+#  Loki
+#  Simple IOC Scanner
+
+import sys
+import hashlib
+import binascii
+import pylzma
+import zlib
+import struct
+import socket
+import traceback
+import os
+import re
+import psutil
+try:
+    from io import StringIO

```

Comment:
This makes it python2 incompatible, I believe.  It was compatible with both before.

Reply:
I left it there but StringIO is not used in the code, because in py3 you should use BytesIO, can be an option using both to assure py2 compatability

Reply:
My point here is that StringIO is located in different modules between the two python versions.  Your change makes it so it tries io, and if that fails it just tries io again, which we can expect to fail again.  By trying to import it from two different modules, as it was before, it should work for both py2 and py3.
The StringIO name isn't used in helpers.py, but is used within loki.py itself (without begin imported there, I assume because it's imported here).  That's pretty hackish itself, but that's apparently how it was hacked together...